src/main/java/org/embulk/output/dynamodb/DynamodbOutputPlugin.java

```diff
@@ -285,14 +292,51 @@ public void jsonColumn(Column column)
                                 addNullValue(column.getName());
                             }
                             else {
-                                item.withJSON(column.getName(), pageReader.getJson(column).toString());
+                                Value jsonValue = pageReader.getJson(column);
+                                if (jsonValue.isArrayValue()) {
+                                    List<Object> list = new ArrayList<Object>();
+                                    for (Value v : jsonValue.asArrayValue()) {
+                                        list.add(getRawFromValue(v));
+                                    }
+                                    item.withList(column.getName(), list);
+                                } else {
+                                    item.withJSON(column.getName(), jsonValue.toJson());
+                                }
                             }
                         }
 
                         private void addNullValue(String name)
                         {
                             item.withNull(name);
                         }
+
+                        private Object getRawFromValue(Value value)
+                        {
+                            if (value.isBooleanValue()) {
+                                return ((BooleanValue)value).getBoolean();
+                            } else if (value.isStringValue()) {
+                                return value.toString();
+                            } else if (value.isIntegerValue()) {
+                                return ((IntegerValue)value).asLong();
+                            } else if (value.isFloatValue()) {
+                                return ((FloatValue)value).toDouble();
+                            } else if (value.isArrayValue()) {
+                                List<Object> list = new ArrayList<>();
+                                for (Value v : value.asArrayValue()) {
+                                    list.add(getRawFromValue(v));
+                                }
+                                return list;
+                            } else if (value.isMapValue()) {
+                                Map<String, Object> map = new LinkedHashMap<>();
+                                for (Map.Entry<Value, Value> entry : value.asMapValue().entrySet()) {
+                                    map.put(entry.getKey().toString(), getRawFromValue(entry.getValue()));
+                                }
+                                return map;
+                            } else if (value.isNilValue()) {
+                                return null;
+                            }
+                            throw new RuntimeException("UnSupported Type");
```

Comment:
Thank you for fixing.
I have 1 comment.

I think plugin should throw  [DataException](https://github.com/embulk/embulk/blob/master/embulk-standards/src/main/java/org/embulk/standards/CsvParserPlugin.java#L368) or its [subclass](https://github.com/hakobera/embulk-input-mongodb/blob/master/src/main/java/org/embulk/input/mongodb/ValueCodec.java#L174) instead of RuntimeException.

Here is a principle
* plugin should throw those type Exception if finds an invalid record.
* plugin should throw ConfigException or its subclass if user configuration is invalid.


Everything else looks good to me 👍